src/test/javascript/spec/component/text/manual-text-selection.component.spec.ts

```diff
@@ -0,0 +1,73 @@
+import * as chai from 'chai';
+import * as sinonChai from 'sinon-chai';
+import * as sinon from 'sinon';
+import { TestBed, ComponentFixture } from '@angular/core/testing';
+import { ArtemisTestModule } from '../../test.module';
+import { ArtemisSharedModule } from 'app/shared/shared.module';
+import { ArtemisConfirmIconModule } from 'app/shared/confirm-icon/confirm-icon.module';
+import { MockTranslateService } from '../../helpers/mocks/service/mock-translate.service';
+import { MockComponent } from 'ng-mocks';
+import { FaIconComponent } from '@fortawesome/angular-fontawesome';
+import { TranslateService } from '@ngx-translate/core';
+import { ManualTextSelectionComponent } from 'app/exercises/text/shared/manual-text-selection/manual-text-selection.component';
+import { SelectionRectangle, TextSelectEvent } from 'app/exercises/text/shared/text-select.directive';
+
+chai.use(sinonChai);
+const expect = chai.expect;
+
+describe('ManualTextSelectionComponent', () => {
+    let component: ManualTextSelectionComponent;
+    let fixture: ComponentFixture<ManualTextSelectionComponent>;
+
+    beforeEach(async () => {
+        TestBed.configureTestingModule({
+            imports: [ArtemisTestModule, ArtemisSharedModule, ArtemisConfirmIconModule],
+            declarations: [ManualTextSelectionComponent],
+            providers: [{ provide: TranslateService, useClass: MockTranslateService }],
+        })
+            .overrideModule(ArtemisTestModule, {
+                remove: {
+                    declarations: [MockComponent(FaIconComponent)],
+                    exports: [MockComponent(FaIconComponent)],
+                },
+            })
+            .compileComponents();
+    });
+
+    beforeEach(() => {
+        fixture = TestBed.createComponent(ManualTextSelectionComponent);
+        component = fixture.componentInstance;
+        fixture.detectChanges();
+    });
+
+    it('should select text and assess the text', () => {
+        const rectangle = { left: 0, top: 0, width: 0, height: 0 } as SelectionRectangle;
+        const event = { text: 'This is a text\n another line', viewportRectangle: null, hostRectangle: null } as TextSelectEvent;
+
+        component.disabled = true;
+
+        // catch edge case
+        component.didSelectSolutionText(event);
```

Comment:
Is this missing an `expect`? (We overwrite the values immediately in the next lines.)

Reply:
To achieve 100% coverage, i wanted to cover the edge case when disabled is set to true. It does not set any data so nothing can be expected here

Reply:
Ah ok.